includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1267,6 +1267,11 @@ public function process_payment_for_order( $cart, $payment_information, $additio
 		if ( $payment_needed ) {
 			$payment_method_details = $intent->get_payment_method_details();
 			$payment_method_type    = $payment_method_details ? $payment_method_details['type'] : null;
+
+			if ( $payment_method_details && 'card' === $payment_method_details['type'] ) {
+				$order->add_meta_data( 'last4', $payment_method_details['card']['last4'] );
+				$order->save_meta_data();
```

Comment:
Possibly related to the above comment, this adds a `last4` metadata item for all orders paid with credit card, even if the order isn't paid with WooPay. If we only need it for WooPay orders, can/should we avoid adding the metadata for non-WooPay orders?